src/code/audio_seqplayer.c

```diff
@@ -956,11 +976,797 @@ void Audio_SequenceChannelSetVolume(SequenceChannel* seqChannel, u8 volume) {
+                        channel->unk_0C = 0;
+                        channel->adsr.sustain = 0;
+                        channel->velocityRandomVariance = 0;
+                        channel->durationRandomVariance = 0;
+                        channel->unk_0F = 0;
+                        channel->unk_20 = 0;
+                        channel->bookOffset = 0;
+                        channel->freqScale = 1.0f;
+                        break;
+                    case 0xE9:
+                        func_800EAEF4(channel, (u8)parameters[0]);
+                        break;
+                    case 0xED:
+                        command = (u8)parameters[0];
+                        channel->unk_0C = command;
+                        break;
+                    case 0xB0:
+                        offset = (u16)parameters[0];
+                        data = player->seqData + offset;
+                        channel->unk_CC = (s16*)data;
+                        break;
+                    case 0xB1:
+                        channel->unk_CC = NULL;
+                        break;
+                    case 0xB3:
+                        command = parameters[0];
+
+                        if (channel->unk_CC != NULL) {
+                            lowBits = (command >> 4) & 0xF;
+                            command &= 0xF;
+                            func_800DF688(channel->unk_CC, lowBits, command);
+                        }
+                        break;
+                    case 0xB2:
+                        offset = (u16) parameters[0];
+                        channel->pad_22 = *(u16*)(offset + scriptState->value * 2 + player->seqData);
+                        break;
+                    case 0xB4:
+                        channel->dynTable = (void*)&player->seqData[channel->pad_22];
+                        break;
+                    case 0xB5:
+                        channel->pad_22 = ((u16*)(channel->dynTable))[scriptState->value];
+                        break;
+                    case 0xB6:
+                        data = (*channel->dynTable)[0];
+                        scriptState->value = (*channel->dynTable)[0][scriptState->value];
+                        break;
+                    case 0xB7:
+                        channel->pad_22 = (parameters[0] == 0) ? gAudioContext.gAudioRandom // odd load here
+                                                                : gAudioContext.gAudioRandom % parameters[0];
+                        break;
+                    case 0xB8:
+                        scriptState->value = (parameters[0] == 0) ? gAudioContext.gAudioRandom
+                                                                : gAudioContext.gAudioRandom % parameters[0];
+                        break;
+                    case 0xBD: {
+                        result = Audio_NextRandom();
+                        channel->pad_22 = (parameters[0] == 0) ? (u32) result : (u32)result % parameters[0];
```

Comment:
I haven't found a vanilla seq that uses a B1-F command yet, so I don't know what this really does, but this is clearly not a pad variable, right? Should it at least get renamed to unk22?